lib/ansible/modules/web_infrastructure/ansible_tower/tower_credential.py

```diff
@@ -165,6 +165,43 @@
     pass
 
 
+KIND_CHOICES = {
+    'ssh': 'Machine',
+    'vault': 'Ansible Vault',
+    'net': 'Network',
+    'scm': 'Source Control',
+    'aws': 'Amazon Web Services',
+    'vmware': 'VMware vCenter',
+    'satellite6': 'Red Hat Satellite 6',
+    'cloudforms': 'Red Hat CloudForms',
+    'gce': 'Google Compute Engine',
+    'azure_rm': 'Microsoft Azure Resource Manager',
+    'openstack': 'OpenStack',
+    'rhv': 'Red Hat Virtualization',
+    'insights': 'Insights',
+    'tower': 'Ansible Tower',
+}
+
+
+def credential_type_for_v1_kind(params, module):
+    credential_type_res = tower_cli.get_resource('credential_type')
+    kind = params.pop('kind')
+    arguments = {'managed_by_tower': True}
+    if kind == 'ssh':
+        if params.get('vault_password'):
+            arguments['kind'] = 'vault'
+        else:
+            arguments['kind'] = 'ssh'
```

Comment:
So if a credential specifies both a vault password and an ssh key, we can't really help them because these split into 2 credentials in the migration. But the credential lookup will be by name... so hopefully vault is the one that kept the old name in these cases?

Reply:
https://github.com/ansible/awx/blob/9dbcc5934ee1a5774cec58b8f4ce08bea777dd7e/awx/main/migrations/_credentialtypes.py#L126

I can't figure out what happened to the names in this case.

Oh, but credentials are `CommonModelNameNotUnique`. So the one thing that's clear is that this case is pretty hopeless.

Reply:
Yep, that's probably about the best we can do - this is what the v1 compat shim in the API itself does.